cilium-cli/encrypt/encrypt.go

```diff
@@ -54,5 +55,8 @@ func (s *Encrypt) fetchCiliumPods(ctx context.Context) ([]corev1.Pod, error) {
 	if err != nil {
 		return nil, fmt.Errorf("unable to list Cilium pods: %w", err)
 	}
+	if len(pods.Items) == 0 {
+		return nil, errors.New("unable to find Cilium pods")
+	}
```

Comment:
I'm guessing many parts of the CLI use this function. Are we sure we're not going to break stuff with this change in the function's behavior? Why is this change needed?

Reply:
This method is used only in 1 place: https://github.com/cilium/cilium/pull/37018/files#diff-73920c9c21ba8049dfa165ca3e9f5cf0e05ca9b84c80932b2a95937bcff0d411R31
I think it doesn't make sense to continue if there are no Cilium pods.

Reply:
Oh, I see. `Encrypt` has its own `fetchCiliumPods`. Sorry for the noise!